extensions/resume.md

```diff
@@ -0,0 +1,306 @@
+
+
+### Capabilities
+The `draft/resume-0.4` capability is advertised by servers that support resuming connections.
+
+When this capability is negotiated, the server sends the client their resume token, which is used to authenticate future connection resumptions. Servers MAY extend the "ping timeout" length for that client (with the expectation that if the client disconnects, it will try to resume its connection as described here). Servers MAY also treat disconnections without `QUIT` as though the client used the `BRB` command described below.
+
+Servers MUST ONLY generate and provide a resume token to a client when the client negotiates the `draft/resume-0.4` capability.
+
+Capability negotiation example:
+
+    C: CAP LS
+    S: CAP * LS :multi-prefix draft/resume-0.4
+    C: CAP REQ :draft/resume-0.4
+    S: CAP * ACK :draft/resume-0.4
+    S: RESUME TOKEN JKbAypzFiovffzuD8VEfcs6bOLrXsSenxsyZNt87Bb0YaQlyOzuQciz6cB2R
+
+
+### Resuming Messages
+
+#### `RESUME` Command
+This command, sent from a client to a server, indicates that the client wishes to resume their old session. This command MAY ONLY be sent during connection registration.
+
+    RESUME <token> [timestamp]
+
+`<token>` is the old connection's resume token.
+
+`<timestamp>`, if given, is a timestamp indicating when the client received the last message from the server on the old connection. This timestamp uses the same format as the IRCv3 `server-time` extension (i.e. `YYYY-MM-DDThh:mm:ss.sssZ`, or in UTC using extended format as specified by ISO 8601:2004(E) 4.3.2), and is passed to other clients to indicates how long the disconnection lasted.
+
+If the request is successful, the server returns a `RESUME` message as described below, registration immediately completes, and the server begins replaying the current client state.
+
+If the request is unsuccessful, the server returns a `FAIL RESUME` message with one of the codes below using the given format, including an appropriate description of the error:
+
+| Code | Format |
+| ---- | ------ |
+| `INSECURE_SESSION` | `:<server> FAIL RESUME INSECURE_SESSION :Cannot resume connection, you are not connected with TLS` |
+| `INVALID_TOKEN` | `:<server> FAIL RESUME INVALID_TOKEN :Cannot resume connection, token is not valid` |
+| `REGISTRATION_IS_COMPLETED` | `:<server> FAIL RESUME REGISTRATION_IS_COMPLETED :Cannot resume connection, connection registration has completed` |
+| `CANNOT_RESUME` | `:<server> FAIL RESUME CANNOT_RESUME :Cannot resume connection, for a different reason described here` |
+
+If a client receives a `FAIL RESUME` message, regardless of the code, then they MUST abort the resume attempt and connect to the server normally instead.
+
+#### `RESUME` Message
+This message, sent from a server to a client, indicates that a `RESUME` request has been successful.
+
+    RESUME <oldnick>
```

Comment:
When compared to the server giving us a new RESUME token during CAP negotiation...

> RESUME TOKEN JKbAypzFiovffzuD8VEfcs6bOLrXsSenxsyZNt87Bb0YaQlyOzuQciz6cB2R

What if a user's nickname was `TOKEN`? Seems clumsy to use use `RESUME <nickname>` because of this.